arviz/utils.py

```diff
@@ -100,3 +100,66 @@ def format_sig_figs(value, default=None):
     if value == 0:
         return 1
     return max(int(np.log10(np.abs(value))) + 1, default)
+
+
+def conditional_vect(function=None, **kwargs):  # noqa: D202
+    """Use numba's vectorize decorator if numba is installed.
+
+    Notes
+    -----
+        If called without arguments  then return wrapped function.
+        @conditional_vect
+        def my_func():
+            return
+        else called with arguments
+        @conditional_vect(nopython=True)
+        def my_func():
+            return
+
+    """
+
+    def wrapper(function):
+        try:
+            numba = importlib.import_module("numba")
+            return numba.vectorize(**kwargs)(function)
+
+        except ImportError:
+            return function
+
+    if function:
+        return wrapper(function)
+    else:
+        return wrapper
+
+
+def numba_check():
+    """Check if numba is installed."""
+    numba = importlib.util.find_spec("numba")
+    return numba is not None
+
+
+class Numba:
+    """A class to toggle numba states."""
+
+    numba_flag = numba_check()
+
+    @classmethod
+    def disable_numba(cls):
+        """To disable numba."""
+        cls.numba_flag = False
+
+    @classmethod
+    def enable_numba(cls):
+        """To enable numba."""
+        if numba_check():
+            cls.numba_flag = True
+        else:
+            raise ValueError("Numba is not installed")
+
+
+def _numba_var(function_1, function_2, data, axis=None, ddof=0):
+    """To replace the if-else blocks."""
```

Comment:
Slight nudge here to change the function_1, function_2 to something more human recognizable like "numba_func", "numpy_func" or something like that, and expand out docstring

I would also change the docstring from 

`     """To replace the if-else blocks."""`
because this only makes sense to us from history. It won't make sense to someone in the future because they can't see the if else blocks that were replaced.

Something like 

> Returns appropriate function for numba speedup if Numba is installed or enabled

With argument documentation as well.